src/server/game/Spells/Spell.cpp

```diff
@@ -2287,6 +2287,10 @@ void Spell::DoAllEffectOnTarget(TargetInfo* target)
     if (!caster)
         return;
 
+    // Check for SPELL_ATTR7_INTERRUPT_ONLY_NONPLAYER before checking the miss conditions
+    if (m_spellInfo->HasAttribute(SPELL_ATTR7_INTERRUPT_ONLY_NONPLAYER) && unit->GetTypeId() != TYPEID_PLAYER)
```

Comment:
technically it shouldnt check for it to be immuned 
Stuns the target for $d and interrupts non-player spellcasting for $32747d.
So yeah shauren is likely right. Let me move it after the handling the effect.